components/coding/studentPortal/ProgressBar.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+export type ProgressBarProps = {
+  completed: number;
+};
+
+export const ProgressBar: React.FC<ProgressBarProps> = ({
+  completed,
+}: ProgressBarProps) => {
+  const fillerStyles = {
+    height: "100%",
+    width: `${completed}%`,
+    borderRadius: "inherit",
+  };
+  return (
+    <div>
+      <div className="h-8 w-full bg-gray-300 m-32 border rounded-full">
```

Comment:
im not sure that you want to include the margin here. You may want put it in the parent instead, so the progress comes with no margin, but the parent that wants to plug it in can provide the right amount of margin